server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsResponse.java

```diff
@@ -88,12 +90,42 @@ public ClusterStatsResponse(
         List<ClusterStatsNodeResponse> nodes,
         List<FailedNodeException> failures,
         ClusterState state
+    ) {
+        this(
+            timestamp,
+            clusterUUID,
+            clusterName,
+            nodes,
+            failures,
+            state,
+            ClusterStatsRequest.Metric.allMetrics(),
+            ClusterStatsRequest.IndexMetrics.allIndicesMetrics()
+        );
+    }
+
+    public ClusterStatsResponse(
+        long timestamp,
+        String clusterUUID,
+        ClusterName clusterName,
+        List<ClusterStatsNodeResponse> nodes,
+        List<FailedNodeException> failures,
+        ClusterState state,
+        Set<String> requestedMetrics,
+        Set<String> indicesMetrics
     ) {
         super(clusterName, nodes, failures);
         this.clusterUUID = clusterUUID;
         this.timestamp = timestamp;
-        nodesStats = new ClusterStatsNodes(nodes);
-        indicesStats = new ClusterStatsIndices(nodes, MappingStats.of(state), AnalysisStats.of(state));
+        nodesStats = !requestedMetrics.isEmpty() && !requestedMetrics.equals(Set.of(Metric.INDICES.metricName()))
```

Comment:
this is redundant - !requestedMetrics.isEmpty()

Reply:
Tracking this as part of https://github.com/opensearch-project/OpenSearch/pull/15938#discussion_r1775193908